### DISCUSSION TITLE
key bindings in plot widgets

### DISCUSSION THREAD

**Discussion by g3grau** [pyvista/pyvista#4372] [Q&A]
Hi,

I'm trying to get warm with pyvista for a Qt application.
 
There are a couple of plotting demos, e.g. showing a sphere where the camera view can be changed. I figured out that there are some key bindings/hotkeys defined, but I find them by random. I couldn't find any documentation about these bindings or how to modify/add to them. For some I'm clueless what they do:
E.g. s=solid, w=wireframe, v=default view??, i=toggle xyz axes indicator, r=reset zoom/pan??, and some more. I'm clueless about f and 3 (which paints everything in pink), and clueless why p=bounding box?? can be enable only in s mode and disabled only in w mode.

I guess this may be all defined within VTK rather than pyvista, but in both places I didn't find documentation. Any hints where this is documented or defined?

thanks
  GG

**Comment by g3grau** [comment]
I just came across the definition: [plotting](https://docs.pyvista.org/version/stable/api/plotting/plotting.html)
My search keywords such as bindkeys or key bindings or hotkeys were wrong, keyboard shortcut might have been better.
I also came across a code snippet to add a bindkey in [4195](https://github.com/pyvista/pyvista/discussions/4195), in short:
```
def toggle(*args):
    widget.SetEnabled(not widget.GetEnabled())
pl.add_key_event('b', toggle)

```

Would be still interesting to know where this is defined and if there is a way to modify these defaults

**Comment by MatthewFlamm** [comment]
The default key bindings are mostly due to the vtk default.  The defaults are here https://docs.pyvista.org/api/plotting/plotting.html#plotting, but this list is only accurate for the default interactor style (This needs to be fixed in the documentation).  I don't know of any good way to find the behavior without either trial and error or going to the vtk source code, which is hard to parse for this functionality.

Also see #6654 for modifying the default behavior.
